sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -243,12 +214,14 @@ namespace Azure { namespace Data { namespace Tables {
     /**
      * @brief Submits a transaction.
      *
-     * @param steps the transaction steps to execute.
+     * @param steps The transaction steps to execute.
+     * @param batchId The batchId for the transaction.
      * @param context for canceling long running operations.
      * @return Submit transaction result.
      */
     Response<Models::SubmitTransactionResult> SubmitTransaction(
         std::vector<Models::TransactionStep> const& steps,
+        Azure::Core::Uuid const& batchId = Azure::Core::Uuid::CreateUuid(),
```

Comment:
I don't believe we've ever exposed a Uuid input/output parameter within the SDK, as an exchange type, but rather accept input as `std::string` (similar to `Azure::Core::Url`).

Search for `Uuid` in `*.hpp` within the repo and you'll basically find no results in public surface area except for 1 place to interop in with AmqpValue.

Similarly, look at many of the existing `Id` named input/output parameters in the SDK. They are UUIDs but exposed as std::string.
https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp#L111-L114

@LarryOsterman it would be worth codifying our guideline for this, because previously the decision was to keep the exchange type as string for interoperability with existing code)

Reply:
What does the typespec/swagger for Tables say about the batchId field? Is it a Uuid, or is it a string?

If the typespec/swagger says it is a Uuid, then IMHO we should express this as a Uuid because that is the contract for this field (we had this discussion a while ago when the generator implemented Uuid values - IMHO, if the spec says that it's a Uuid, the API surface should express it as a Uuid).

But looking at the use here (it appears to just be an ID, which typically just needs to be unique), it makes a ton of sense for it to be a string, and not a Uuid.